src/test/scala/org/broadinstitute/hail/variant/IntervalSuite.scala

```diff
@@ -0,0 +1,167 @@
+
+    val ex2 = IntervalListAnnotator.read("src/test/resources/example2.interval_list", hadoopConf)
+    assert(ex1 == ex2)
+  }
+
+  @Test def testNew() {
+    val a = Array(
+      genomicInterval("1", 10, 20),
+      genomicInterval("1", 30, 40),
+      genomicInterval("2", 40, 50))
+
+    val t = IntervalTree[Locus](a)
+
+    assert(t.contains(Locus("1", 15)))
+    assert(t.contains(Locus("1", 30)))
+    assert(!t.contains(Locus("1", 40)))
+    assert(!t.contains(Locus("1", 20)))
+    assert(!t.contains(Locus("1", 25)))
+    assert(!t.contains(Locus("1", 9)))
+    assert(!t.contains(Locus("5", 20)))
+
+    // Test queries
+    val a2 = Array(
+      genomicInterval("1", 10, 20),
+      genomicInterval("1", 30, 40),
+      genomicInterval("1", 30, 50),
+      genomicInterval("1", 29, 31),
+      genomicInterval("1", 29, 30),
+      genomicInterval("1", 30, 31),
+      genomicInterval("2", 41, 50),
+      genomicInterval("2", 43, 50),
+      genomicInterval("2", 45, 70),
+      genomicInterval("2", 42, 43),
+      genomicInterval("2", 1, 10))
+
+    val t2 = IntervalTree[Locus](a2)
+
+    assert(t2.query(Locus("1", 30)) == Set(
+      genomicInterval("1", 30, 40), genomicInterval("1", 30, 50),
+      genomicInterval("1", 29, 31), genomicInterval("1", 30, 31)))
+
+  }
+
+  @Test def properties() {
+
+    // greater chance of collision
+    val lgen = Gen.zip(Gen.oneOf("1", "2"),
+      Gen.choose(1, 100))
+    val g = Gen.zip(IntervalTree.gen(lgen),
+      lgen)
+
+    val p = Prop.forAll(g) { case (it, locus) =>
+      val intervals = it.toSet
+
+      val setResults = intervals.filter(_.contains(locus))
+      val treeResults = it.query(locus)
+
+      val inSet = intervals.exists(_.contains(locus))
+      val inTree = it.contains(locus)
+
+      setResults == treeResults &&
+        setResults.nonEmpty == inSet &&
+        inSet == inTree
+    }
+    p.check()
+  }
+
+  @Test def testAggregate() {
+    val vds = LoadVCF(sc, "src/test/resources/sample2.vcf", nPartitions = Some(2)).cache()
+    val state = SplitMulti.run(State(sc, sqlContext, vds))
+    val iList = tmpDir.createTempFile("input", ".interval_list")
+    val tmp1 = tmpDir.createTempFile("output", ".tsv")
+
+    val startPos = 16050036 - 250000
+    val endPos = 17421565 + 250000
+    val g = Gen.parameterized { p =>
+      val rng = p.rng
+      Gen.const((0 until 500).map { _ =>
+        val start = rng.nextInt(startPos, endPos)
+        val end = rng.nextInt(start, endPos)
+        Interval(Locus("22", start), Locus("22", end))
+      })
+    }
+    val p = Prop.forAll(g) { intervals =>
+
+      writeTextFile(iList, hadoopConf) { out =>
+        intervals.foreach { i =>
+          out.write(s"22\t${i.start.position}\t${i.end.position}\n")
+        }
+      }
+
+      AggregateIntervals.run(state, Array(
+        "-i", iList,
+        "-o", tmp1,
+        "-c", "nSNP = variants.count(v.altAllele.isSNP), nIndel = variants.count(v.altAllele.isIndel), N = variants.count(true)"))
```

Comment:
Way to use the example in the test, awesome.  Update this with `va` and `gs` when you make that change.